File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.py
class Solution:
    def countOrders(self, n: int) -> int:
        places, cnt, mod = 2*n, 1, 1e9+7
        for i in range(n, 0, -1):
            cnt = (cnt*((places*(places-1))//2))%mod
            places -= 2
        return int(cnt)

File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.java
class Solution {
    public int countOrders(int n) {
        int mod = 1000000007;
        long places = 2*n, cnt = 1;
        for(int i=n; i>=1; i--){
            cnt = (cnt*((places*(places-1))/2))%mod;
            places -= 2;
        }
        return (int)cnt;
    }
}